Algorithm.CSharp/AltData/BenzingaNewsAlgorithm.cs

```diff
@@ -0,0 +1,93 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System.Collections.Generic;
+using System.Linq;
+using QuantConnect.Algorithm.Framework.Selection;
+using QuantConnect.Data;
+using QuantConnect.Data.Custom.Benzinga;
+using QuantConnect.Data.UniverseSelection;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    public class BenzingaNewsAlgorithm : QCAlgorithm
+    {
+        /// <summary>
+        /// Initialise the data and resolution required, as well as the cash and start-end dates for your algorithm. All algorithms must initialized.
+        /// </summary>
+        public override void Initialize()
+        {
+            SetStartDate(2018, 10, 12);
+            SetEndDate(2018, 11, 25);
+            SetCash(100000);
+
+            UniverseSettings.Resolution = Resolution.Daily;
+            AddUniverseSelection(new CoarseFundamentalUniverseSelectionModel(CoarseSelector));
+        }
+
+        public IEnumerable<Symbol> CoarseSelector(IEnumerable<CoarseFundamental> coarse)
+        {
+            // Add Benzinga news data from the filtered coarse selection
+            var symbols = coarse.Where(x => x.HasFundamentalData && x.DollarVolume > 50000000)
+                .Select(x => x.Symbol)
+                .Take(10);
+
+            foreach (var symbol in symbols)
+            {
+                AddData<BenzingaNews>(symbol);
+            }
+
+            return symbols;
+        }
+
+        public override void OnData(Slice data)
+        {
+            // Get all Benzinga data and loop over it
+            foreach (var article in data.Get<BenzingaNews>().Values)
+            {
+                // Select the same Symbol we're getting a data point for
+                // from the articles list so that we can get the sentiment of the article.
+                // We use the underlying Symbol because the Symbols included in the `Symbols` property
+                // are equity Symbols.
+                var selectedSymbol = article.Symbols.SingleOrDefault(x => x.Symbol == article.Symbol.Underlying);
+                if (selectedSymbol == null)
+                {
+                    continue;
```

Comment:
Is this possible? if this `article` instance is for the `Symbol.Underlying` why wouldn't it be present? seems like a bug in the data converter/source?